src/bin/add/main.rs

```diff
@@ -141,7 +145,34 @@ fn handle_add(args: &Args) -> Result<()> {
         )?;
         update_registry_index(&url, args.quiet)?;
     }
-    let deps = &args.parse_dependencies()?;
+    let requested_features: Option<BTreeSet<&str>> = args.features.as_ref().map(|v| {
+        v.iter()
+            .map(|s| s.split(' '))
+            .flatten()
+            .filter(|s| !s.is_empty())
+            .collect()
+    });
+
+    let deps = &args.parse_dependencies(
+        requested_features
+            .as_ref()
+            .map(|s| s.iter().map(|s| s.to_string()).collect()),
+    )?;
+
+    if let Some(req_feats) = requested_features {
+        assert!(deps.len() == 1);
+        let available_features = deps[0]
+            .available_features
+            .iter()
+            .map(|s| s.trim())
```

Comment:
Do we need this trim?

Reply:
I'm actually not sure if `toml_edit` does that for us. I'd assume so, in which case you are correct that we don't, so I'll just remove it.

Reply:
Oh, this is on available features.

... I actually don't know if whitespace is tolerated or not in feature names.  Looks like it isn't
```
❯ cargo check
warning: invalid character ` ` in feature `foo ` in package cargo-edit v0.8.0 (/home/epage/src/personal/cargo-edit), characters must be Unicode XID characters, `+`, or `.` (numbers, `+`, `-`, `_`, `.`, or most letters)
This was previously accepted but is being phased out; it will become a hard error in a future release.
For more information, see issue #8813 <https://github.com/rust-lang/cargo/issues/8813>, and please leave a comment if this will be a problem for your project.
warning: invalid character ` ` in feature `foo ` in package cargo-edit v0.8.0 (/home/epage/src/personal/cargo-edit), characters must be Unicode XID characters, `+`, or `.` (numbers, `+`, `-`, `_`, `.`, or most letters)
This was previously accepted but is being phased out; it will become a hard error in a future release.
For more information, see issue #8813 <https://github.com/rust-lang/cargo/issues/8813>, and please leave a comment if this will be a problem for your project.
```

So we shouldn't need a trim